tests/responses/test_fileresponse.py

```diff
@@ -80,3 +80,51 @@ def test_unicode_attachment(self):
             response['Content-Disposition'],
             "attachment; filename*=utf-8''%E7%A5%9D%E6%82%A8%E5%B9%B3%E5%AE%89.odt"
         )
+
+    def test_file_to_stream_closes_response(self):
+        # Closing file_to_stream calls FileResponse.close(), even when
+        # file-like object doesn't have a close() method.
+        class FileLike:
+            def read(self):
+                pass
+
+        class FileLikeWithClose(FileLike):
+            def __init__(self):
+                self.closed = False
+
+            def close(self):
+                self.closed = True
+
+        for filelike_cls in (FileLike, FileLikeWithClose):
+            with self.subTest(filelike_cls=filelike_cls.__name__):
+                filelike = filelike_cls()
+                response = FileResponse(filelike)
+                file_to_stream = response.file_to_stream
+                self.assertFalse(response.closed)
+                file_to_stream.close()
+                # Object with close() is added to the list of closable.
+                if hasattr(filelike, 'closed'):
+                    self.assertEqual(response._closable_objects, [filelike])
+                    self.assertTrue(filelike.closed)
+                else:
+                    self.assertEqual(response._closable_objects, [])
+                self.assertTrue(response.closed)
+
+    def test_file_to_stream_closes_response_on_error(self):
+        # Closing file_to_stream calls FileResponse.close(), even when
+        # closing file-like raises exceptions.
+        class FileLikeWithRaisingClose:
+            def read(self):
+                pass
+
+            def close(self):
+                raise RuntimeError()
+
+        filelike = FileLikeWithRaisingClose()
+        response = FileResponse(filelike)
+        file_to_stream = response.file_to_stream
+        self.assertFalse(response.closed)
+        with self.assertRaises(RuntimeError):
+            file_to_stream.close()
+        self.assertEqual(response._closable_objects, [filelike])
```

Comment:
Similarly here, I think `_closable_objects` should be checked after constructing the `response` object and before calling `close()`, as I had it.